### ISSUE TITLE
EDITCALLERS (FINDCALLERS) breaks when  a file has unknown package

### ISSUE THREAD

**Issue by rmkaplan** [Interlisp/medley#130]
**Describe the bug**
(EDITCALLERS 'FOO 'LIBARY/SEDIT-COMMONLISP) stops with a message that it can't find the TM (textmodule?) package, which is in some other file, apparently not included in the FULL sysout.

**To Reproduce**
1. Files loaded '...'
2. Form to eval and/or menus to click '....'
3. What happened? '....'

**Expected behavior**

**Screenshots**

**Context (please complete the following information):**
 - OS: [e.g. Mac/Linux/Cygwin]
 - OS Version: [e.g. High Siera/Ubuntu 18/Raspbian]
 - Host arch: [e.g. x86_64, arm7l, arm64, sparc]
 - Maiko version: [e.g. commit ID from `git log | head`]
 - IL:MAKESYSDATE: [ date ]

**Additional context**

**Comment by nbriggs** [comment]
I realize the problem report is for the case when you *don't* have the package already existing.
In this case, if you were to load TEXTMODULES.DFASL would it work?

**Comment by rmkaplan** [comment]
Yes, that’s the case.  When you don’t actually know which file you need to load, that’s what you are trying to find out (would be nice to have a masterscope database of the whole system)

Sometimes the error happens in decoding the filecreated expression, but in this sedit example it happens when it is doing something down in the file—executing a load command. 

Wrapping the filecreated parser in an NLSETQ would suppress errors of the first type, an easy fix. Otherwise, a little more investigating to understand why EDITCALLES/FINDCALLERS is executing file commands.

**Comment by masinter** [comment]
this is already covered by https://github.com/Interlisp/medley/issues/31 
does this add any details?
Yes we need a strategy for dealing with packages before loading in things via the filemap.

**Comment by rmkaplan** [comment]
This breaks in an additional way, not just because of the filecreated.  

Wrapping the call to EDITCALLERS Inside FINDCALLERS in an NLSETQ might make the breaks go away, when the point is to do a scan and not actually to do any editing.

**Comment by masinter** [comment]
There's a problem here,but I'm not sure it has much to do with packages. SEDIT-COMMONLISP  refers to TEXTMODULE, which doesn't exist in any version of any Interlisp / Lispcore / Envos. There are lots of TEXTMODULES in Lispusers, Trying to fix this and MAKEFILE(SEDIT-COMMONLISP) led to other problems. Needs more investigation when I can keep my eyes open.

**Comment by masinter** [comment]
![Screenshot 2021-01-04 220129](https://user-images.githubusercontent.com/1116587/103612456-52dcbd80-4ed9-11eb-8b7b-909d0356e5c6.png)
I've assigned this to Ron. My suggestion is to wind back SEDIT-COMMONLISP to 1994 and remove the two versions Ron made on 20-Sep-2020

**Comment by rmkaplan** [comment]
I can do that.  But I think the reason I did this was because of the package problem—it then breaks in the more traditional way.  That apparently triggered something else.  In looking at this, do you see why it was doing the loads, does that make sense?

And actually I think it breaks when you try to do anything else with that file, because it then doesn’t know where to get its package.

**Comment by masinter** [comment]
When you spell the name right, the package is in the full.sysout.
When a change affects what's in the full sysout, doing a loadup and testing it should be done before the PR

**Comment by rmkaplan** [comment]
I see that the typo was a bad fix to patch around that particular instance of the general problem.

Before I made that edit, the file SEDIT-COMMONLISP did not have any reference at all to the source of the TM package that it made use of.  And TEXTMODULES was never in the sysout (in fact, SEDIT-COMMONLISP was also not in the full sysout). And SEDIT-COMMONLISP also wasn’t in the sysout: it was the attempt to do a generic FINDCALLERS directory scan for something else that brought this particular disconnect into focus.

I hunted all through the system to find the TM source (and TM is not a particularly revealing name), and tried to at least have the sedit file have an explicit reference to it.  Even though I don’t know what a textmodule is or even what SEDIT-COMMONLISP is doing.  At least SEDIT-COMMONLISP wouldn’t break in a load.

(By the way, lispusers/TEXTMODULES doesn’t seem to have any documentation, at least not in an obvious place, and it also doesn’t have a compiled counterpart.)

But that doesn’t address the general problem, which is that a generic scanner like FINDCALLERS currently can’t operate on files that reference packages whose sources are not explicit (maybe because the source of that package is a file known to a higher-level loader) and not yet in the current sysout.  And you wouldn’t necessarily want to corrupt the current system by doing an implicit LOAD just because you are poking around with a scan, which also appears to be happening.

So, fixing this particular typo is good, since it now makes SEDIT-COMMONLISP usable independent of scanning.  I suppose that TEXTMODULES should be compiled and documented.

But FINDCALLERS still needs some work.

**Comment by masinter** [comment]
I think the idea is to use GitHub issues as the way to focus discussions. There is the #31 "need better integration of filepkg with packages". which seems like it should cover all of the various ways in which Interlisp interacts with code that hasn't been read in, and also what is necessary to write out in the files to support it.  Interlisp never automated file dependencies, it requires the author of a package to set it up through FILES commands.

there's more of a mystery around TEXTMODULES though, which I'll detail in a new Issue

**Comment by rmkaplan** [comment]
Are you saying that something should be done on the web site rather than replying to messages?

Or just that new issue should be broken out.

**Comment by nbriggs** [comment]
Re --
>(By the way, lispusers/TEXTMODULES doesn’t seem to have any documentation, at least not in an obvious place, and it also doesn’t have a compiled counterpart.)

I see that the lispusers version of TEXTMODULES is out-of-date (24-Sep-1987) compared to the library (and lispcore/library) version (7-Jan-91).  The DFASLs are in library and lispcore/library.

I'm continuing to maintain my original copy of the medley3.5 directory as I still believe that for the most part the lispcore directory was the truth copy, and the parallel library and lispusers directories there were the derived install directories.
